### DISCUSSION TITLE
Fix The "'nopython' keyword argument was not supplied to the 'numba.jit' decorator." Warning.

### DISCUSSION THREAD

**Discussion by rudymohammadbali** [openai/whisper#1409] [General]
Every time you run the whisper module it shows this warning: 

```
D:\Coding\envs\STT\lib\site-packages\whisper\timing.py:58: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
```

### **Solve the warning**

Add **nopython=True** inside **@numba.jit()** decorator in file ***\lib\site-packages\whisper\timing.py:58**

Like this:

`@numba.jit(nopython=True)`

**Comment by TrigMan59** [comment]
I get the same error message. I realize this is a complete newbie question but where exactly can I find this file and how can I edit it?

**Reply by phineas-pta** [reply]
that's a warning not a error so you don't really need to do anything

if it bothers you much and you want it to disappear, you have to identify pythin packages directory on your pc, edit the source code of whisper, clear the pycache and rerun the code

**Reply by rudymohammadbali** [reply]
> I get the same error message. I realize this is a complete newbie question but where exactly can I find this file and how can I edit it?

### It is not a error but it's a warning message from numba module. 

**To fix it just like I said:**

Go to _timing.py file in whisper folder at line 58_ and change **@numba.jit()** to **@numba.jit(nopython=True)**

**Reply by TrigMan59** [reply]
@phineas-pta Thanks for your reply, I appreciate it. I looked in my C:\Users\roger\.cache\whisper folder and there is no file named timing.py

Is it somewhere else on my system? Sorry for the noob question.

**Reply by phineas-pta** [reply]
it's not the whisper cache folder, you have to look for the python packages directory

also you don't have to be sorry for being a noob, but should be for not googling before asking

**Reply by TrigMan59** [reply]
@phineas-pta Thanks so much for your reply. I did try Googling it and I did find 2 timing.py files on my PC. When I opened each one, Line 58 had different code written, so I didn't bother with it. I'll live with the error message for now. Thanks again.

On a totally unrelated note, I wondered if I could ask you a coding question via DM on GitHub. Would this be okay?

**Reply by phineas-pta** [reply]
im not professional programmer so not sure to help you

what is it about?

**Reply by nkabram** [reply]
It's not recommended to edit the site packages directly; it's just a warning; the best is to either suppress the warning for now and wait for an update on the package. It's a relatively new warning, so has patience, and it should be fixed in a soon-to-come update to the whisper package.

**Comment by TrigMan59** [comment]
@phineas-pta Thanks again for the reply. I'm looking for a kind forum where I can get guidance on how to take pieces of GitHub Python scripts here to make a program I really need. Thanks.

**Reply by phineas-pta** [reply]
sorry idk any forum, personally i manually do trial and error until i get right

you may search for some discord / reddit communities

you can just still ask me if your project isn't private, but i'm not sure to be able to answer it
